observation_portal/common/state_changes.py

```diff
@@ -92,36 +91,39 @@ def on_request_state_change(old_request, new_request):
                 modify_ipp_time_from_requests(ipp_value, [new_request], 'credit')
 
 
-@transaction.atomic
-def on_requestgroup_state_change(old_requestgroup, new_requestgroup):
-    if old_requestgroup.state == new_requestgroup.state:
+def on_requestgroup_state_change(old_requestgroup_state, new_requestgroup):
+    if old_requestgroup_state == new_requestgroup.state:
         return
-    valid_request_state_change(old_requestgroup.state, new_requestgroup.state, old_requestgroup)
+    valid_request_state_change(old_requestgroup_state, new_requestgroup.state, new_requestgroup)
     if new_requestgroup.state in TERMINAL_REQUEST_STATES:
-        for r in new_requestgroup.requests.filter(state__in=['PENDING']):
-            r.state = new_requestgroup.state
-            r.save()
+        for request in new_requestgroup.requests.filter(state__iexact='PENDING'):
+            request.state = new_requestgroup.state
+            request.save()
+        # new_requestgroup.requests.select_for_update(of=('self')).filter(state__iexact='PENDING').update(
```

Comment:
Commented out code

Reply:
removed, don't think we need select_for_update here